File: app/src/main/java/com/infy/dagger2implementation/database/MyDataBase.java
package com.infy.dagger2implementation.database;

import android.arch.persistence.room.Database;
import android.arch.persistence.room.RoomDatabase;

import com.infy.dagger2implementation.database.dao.LocationDao;
import com.infy.dagger2implementation.database.model.LocationData;

@Database(entities = {LocationData.class}, version = 1, exportSchema = false)
public abstract class MyDataBase extends RoomDatabase {
    public abstract LocationDao locationDao();

}

File: app/src/main/java/com/infy/dagger2kotlin/database/MyDataBase.kt
package com.infy.dagger2kotlin.database

import android.arch.persistence.room.Database
import android.arch.persistence.room.RoomDatabase
import com.infy.dagger2kotlin.database.dao.LocationDao
import com.infy.dagger2kotlin.database.model.LocationData

@Database(entities = arrayOf(LocationData::class), version = 1, exportSchema = false)
abstract class MyDataBase : RoomDatabase() {
    abstract fun locationDao(): LocationDao

}
